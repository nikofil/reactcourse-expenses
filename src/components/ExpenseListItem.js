import React from 'react'
import moment from 'moment'
import numeral from 'numeral'
import { Link } from 'react-router-dom'

export default function({id, children, amount, createdAt, note}) {
    return (
        <Link className="list-item" to={`/edit/${id}`}>
            <div>
                <h3 className="list-item__title">{children}</h3>
                <span className="list-item__subtitle">{ moment(createdAt).format('MMMM Do YY') }</span>
            </div>
            <h3 className="list-item__data">{ numeral(amount).format('$0,0') }</h3>
        </Link>
    )
}