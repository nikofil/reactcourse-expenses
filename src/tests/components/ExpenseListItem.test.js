import React from 'react'
import { shallow } from 'enzyme'
import ExpenseListItem from '../../components/ExpenseListItem'

test('should render expense item', () => {
    const wrapper = shallow(<ExpenseListItem id="123" amount={456} createdAt={0} note="Hello world">Desc</ExpenseListItem>)
    expect(wrapper).toMatchSnapshot()
})