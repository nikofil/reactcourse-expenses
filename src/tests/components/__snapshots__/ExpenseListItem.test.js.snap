// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense item 1`] = `
<Link
  className="list-item"
  replace={false}
  to="/edit/123"
>
  <div>
    <h3
      className="list-item__title"
    >
      Desc
    </h3>
    <span
      className="list-item__subtitle"
    >
      January 1st 70
    </span>
  </div>
  <h3
    className="list-item__data"
  >
    $456
  </h3>
</Link>
`;
